### ISSUE TITLE
IReturnVoid throws Uncaught (in promise) SyntaxError: Unexpected end of JSON input

### ISSUE THREAD

**Issue by mayerwin** [ServiceStack/servicestack-client#4]
I have a `PUT` service which returns `IReturnVoid`.

```
// @Route("/data/save", "PUT")
export class SaveData implements IReturnVoid
{
    Data: string[];
    createResponse() {}
    getTypeName() { return "SaveData"; }
}
```

Which I call with:

```
const request = new Dtos.SaveData();
request.Data = [""];
this.client.put(request).then(() => {
	alert("saved");
});
```

However it throws: `SyntaxError: Unexpected end of JSON input`. It is the same issue that we had with other non JSON return types. 

```
        return fetch(url, req).then(function (res) {
          if (!res.ok) throw res;
          var x = typeof request.createResponse == 'function' ? request.createResponse() : null;
          if (typeof x === 'string') return res.text().then(function (o) {
            return o;
          });
          var contentType = res.headers.get("content-type");
          if (contentType && contentType.indexOf("application/json") !== -1) {
            return res.json().then(function (o) {
              return o;
            });
          }
          if (x instanceof Uint8Array) {
            if (typeof res.arrayBuffer != 'function') throw new Error("This fetch polyfill does not implement 'arrayBuffer'");
            return res.arrayBuffer().then(function (o) {
              return new Uint8Array(o);
            });
          } else if (x instanceof Blob) {
            if (typeof res.blob != 'function') throw new Error("This fetch polyfill does not implement 'blob'");
            return res.blob().then(function (o) {
              return o;
            });
          }
          return res.json().then(function (o) { //FAILS HERE
            return o;
          });
        }).catch(function (res) {
```

Am I missing something or this is currently unsupported? `IReturnVoid` services are quite common so this is rather surprising.

**Comment by mayerwin** [comment]
(I am using version 0.0.15)

**Comment by mythz** [comment]
Ok this should be resolved with [this commit](https://github.com/ServiceStack/servicestack-client/commit/34c1dfe33fa68c6bf49cc55cdd69f056f5f58cc2) which is available from **0.0.16** that's now on npm.
